### ISSUE TITLE
Calling from another Python program not working

### ISSUE THREAD

**Issue by gitmylo** [facebookresearch/demucs#504]
## 🐛 Bug Report

using the code from the example [Calling from another Python program](https://github.com/facebookresearch/demucs#calling-from-another-python-program), i'm unable to call main(), because it doesn't take any arguments.

Am i missing something or is something wrong? Or is the example incorrect?

## To Reproduce

1. copy the code from [Calling from another Python program](https://github.com/facebookresearch/demucs#calling-from-another-python-program)
2. make sure you have a correctly named file so it should actually run
3. run the example code
4. get a `TypeError: main() takes 0 positional arguments but 1 was given`

## Expected behavior

Demucs should have ran, as the example says, it should have the same effect as `demucs --mp3 --two-stems vocals -n mdx_extra "track with space.mp3"`. But it does not.

## Actual Behavior

`TypeError: main() takes 0 positional arguments but 1 was given` is thrown because main() doesn't take arguments

## Your Environment

The bug occurs no matter the environment, the part causing the error can be recreated like:

```py
def test_func():
  parser = argparse.ArgumentParser()
  parser.add_argument('example')
  args = parser.parse_args()
  print(args.example)

# Attempt to run (it won't be able to since test_func doesn't take arguments)
test_func(['test'])
```

Or with demucs's function using code from [Calling from another Python program](https://github.com/facebookresearch/demucs#calling-from-another-python-program)
```py
import demucs.separate
demucs.separate.main(["--mp3", "--two-stems", "vocals", "-n", "mdx_extra", "track with space.mp3"])
```

- Python and PyTorch version: Python 3.10.6, pytorch 2.0.0
- Operating system and version (desktop or mobile): Windows 10
- Hardware (gpu or cpu, amount of RAM etc.): RTX 3060 12GB, Ryzen 5 5600X, 48GB DDR4 2133MHz

**Comment by CarlGao4** [comment]
Have you installed the latest version? To install, first install git and run `python3 -m pip install -U git+https://github.com/facebookresearch/demucs#egg=demucs`
Or you can wait until 4.0.1 is released and run `python3 -m pip install -U git` if you don't want to install git

**Comment by gitmylo** [comment]
Right, that works. i guess the pypi package isn't updated to this version yet. thanks.

**Comment by casrar** [comment]
I am also having this issue. @CarlGao4 is this confirmed to be a 4.0.0 issue and is resolved by the 4.0.1 version?

**Comment by CarlGao4** [comment]
Yes!
